src/test/java/bio/terra/service/tabulardata/google/BigQueryPdaoTest.java

```diff
@@ -342,6 +349,67 @@ public void snapshotBuilderQuery() throws Exception {
     List<String> searchConceptNames =
         searchConceptsResult.getResult().stream().map(SnapshotBuilderConcept::getName).toList();
     assertThat(searchConceptNames, contains("concept1"));
+
+    getCountResponseTest(dataset);
+    getCountResponseZeroCaseTest(dataset);
+    getCountResponseFuzzyValuesTest(dataset);
+  }
+
+  private void getCountResponseTest(Dataset dataset) {
+    List<SnapshotBuilderCriteria> criteria =
+        List.of(
+            new SnapshotBuilderProgramDataListCriteria()
+                .values(List.of(0))
+                .kind(SnapshotBuilderCriteria.KindEnum.LIST)
+                .id(1),
+            new SnapshotBuilderProgramDataRangeCriteria()
+                .high(1960)
+                .low(1940)
+                .kind(SnapshotBuilderCriteria.KindEnum.RANGE)
+                .id(0),
+            new SnapshotBuilderDomainCriteria()
+                .conceptId(1)
+                .kind(SnapshotBuilderCriteria.KindEnum.DOMAIN)
+                .id(19));
+    testRollupCountsWithCriteriaAndExpected(dataset, criteria, 20);
+  }
+
+  private void getCountResponseZeroCaseTest(Dataset dataset) throws Exception {
```

Comment:
Why can't this function and others be shared with `AzureIntegrationTest`?

Reply:
That is explained here: https://broadworkbench.atlassian.net/browse/DC-1026, but basically it has to do with connected vs integration tests, and the way the assertions differ. 1026 is all about figuring out a code sharing solution.

Reply:
Ok, sweet. I'm glad that's going to be dealt with :)